test/distributed/_sharded_tensor/test_sharded_tensor.py

```diff
@@ -207,21 +203,27 @@ def test_rand(self):
 
     @sandcastle_skip_if(torch.cuda.device_count() < 1, 'CUDA GPU is needed')
     def test_full(self):
-        fill_value = 23
-        expected_dtype = torch.int32
-        tensor_init_params = TensorInitParams(
-            create_op=CreateOp.FULL,
-            fill_value=fill_value,
-            dtype=expected_dtype,
+        fill_value = 23.5
+        tensor_properties = TensorProperties(
+            # tensor's dtype can be inferred from fill_value
+            dtype=None,
```

Comment:
Shouldn't we have also test where we override the dtype here?

Reply:
Great point, added self.assertEqual(torch.float32, local_tensor.dtype) to test it explicitly